src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -203,6 +206,32 @@ public Task<RestoreResult> ExecuteAsync()
                 restoreTime.Elapsed);
         }
 
+        private void ReplayWarningsAndErrors()
+        {
+            var logMessages = _request?.ExistingLockFile?.LogMessages;
```

Comment:
You can avoid the null check later with:
`var logMessages = _request.ExistingLockFile?.LogMessages ?? Enumerable.Empty<IAssetsLogMessage>();`

also, can _request actually be null? I wouldn't expect that